core/test_segment.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+FILE=test_segment
+VIDEOFILE=$FILE.mp4
+GOFILE=$FILE.go
+
+
+echo "package core" > $GOFILE
+echo "// Auto generated by $0 DO NOT EDIT" >> $GOFILE
+echo "var testSegment = []byte{" >> $GOFILE
+# can't pipe .mp4 file - need seekable output
+ffmpeg -loglevel quiet -f lavfi -i color=white:s=48x48 -vframes 5 \
+    -g 2 -pix_fmt yuv420p -profile:v main -preset:v slow \
+    -metadata title='!' -metadata service_provider='!' -f mp4 $VIDEOFILE
```

Comment:
MP4 produces much larger outputs than mpegts because of all the extra metadata that's written out. I'd suggest staying with ts files for the test cases.

Reply:
.ts file generated as you suggested results in segment violation in LPMS's C code (not found where yet, need to compile on Linux machine with GPU)
trying to generate .ts file using same command as I'm using for .mp4 but with .ts output results in `Undefined constant or missing '(' in 'main'` error from ffmpeg

Reply:
Heh, dumb mistake there on my part - h264 is not being generated with that script. Nice find though, responded to https://github.com/livepeer/lpms/issues/185 about it.

Curious, are we still able to decode h264 elementary streams in go-livepeer without adding it to the list of demuxers? If so, this would generate an even smaller file - 61 bytes on my system:

```
ffmpeg -f lavfi -i color=white:s=16x16 -vframes 1  -c:v libx264 -f mpegts - | \
ffmpeg -f mpegts -i pipe:0 -c:v copy -bsf:v 'filter_units=pass_types=1-5|7-15' -f h264 segment.h264
```

It's so well compressed that gzipping actually makes this larger. That means no need to unzip within the code.

If go-livepeer can't demux that file, then we can just use mpegts. Would rather minimize the number of formats on the list.

```
ffmpeg -f lavfi -i color=white:s=16x16 -vframes 1  -c:v libx264 -f mpegts - | \
ffmpeg -f mpegts -i pipe:0 -c:v copy -bsf:v 'filter_units=pass_types=1-5|7-15' -f mpegts -metadata title='!' -metadata service_provider='! - | gzip > segment.ts.gz
```

gzip'd, this yields a 186 byte file. This can be further piped into `xxd`  without any intermediate files hanging around.

The rough equivalent for mp4 is around 464 bytes gzip'd, and it's not pipeable, so it also needs an annoying intermediate file:

```
ffmpeg -f lavfi -i color=white:s=16x16 -vframes 1  -c:v libx264 -f mpegts - | \
ffmpeg -f mpegts -i pipe:0 -c:v copy -bsf:v 'filter_units=pass_types=1-5|7-15' segment.mp4
gzip segment.mp4
```

Reply:
>  decode h264 elementary streams

Tested - didn't work
So I've changed to .ts segment with 5 frames